src/make_trade.ts

```diff
@@ -196,19 +196,27 @@ async function getPotentialBuyTokens(
       continue;
     }
     try {
-      await api.getFee(
+      // Check that a fee path exists to the candidate
+      const fee = await api.getFee(
         sellToken.address,
         buyToken.address,
         amount,
         OrderKind.SELL
       );
+      if (fee.gte(amount)) {
```

Comment:
I'm not sure I understand why if the fee is greater than the amount being traded there isn't enough balance. Should the check be
```suggestion
      if (fee.add(amount).gt(balance)) {
```

Reply:
We intend to trade our complete balance (amount == balance). So if the fee > amount we cannot trade since we don't have enough balance. Will rename the variable to make it more clear.

Reply:
Oh, yeah:
```
const sellAmountAfterFee = sellBalance.sub(fee);
```

and the sell token balance is used as input to this method.

Checks out.